test/integration/systemd_resolved/systemd_resolved_spec.rb

```diff
@@ -0,0 +1,17 @@
+describe file('/etc/systemd/resolved.conf') do
+  its('content') { should match /DNS=1.1.1.1 1.0.0.1 2606:4700:4700::1111 2606:4700:4700::1001/ }
+  its('content') { should match /FallBackDNS=8.8.8.8 8.8.4.4 2001:4860:4860::8888 2001:4860:4860::8844/ }
+  its('content') { should match /DNSSEC=allow-downgrade/ }
+  its('content') { should match /Domains=localtest.com/ }
+  its('content') { should match /CacheFromLocahost=no/ }
+end
+
+describe command('curl https://www.google.com') do
+  its('exit_status') { should eq 0 }
+end
+
```

Comment:
I'd suggest a dig test here, but I think it will return the wrong SERVER value if resolv.conf points to a systemd-resolved stub resolver listening on 127.0.0.1

Reply:
Also, should you try to add a test here, you'll need to either wildcard to match all IPs or only configure one, because systemd will choose them randomly: https://github.com/systemd/systemd/issues/5755

Reply:
The only thing there is that different distro use different stub addresses in the 127.0.0.0/8 subnet so it'll probably have to be a regex.